src/AutorestSwift/Code Generator/SwiftNamer.swift

```diff
@@ -26,21 +26,133 @@
 
 import Foundation
 
+enum SdkRole: String {
+    case service
+    case operationGroup
+    case operation
+    case model
+    case property
+    case parameter
+}
+
 class SwiftNamer: CodeNamer {
     // MARK: Properties
 
     let model: CodeModel
 
     // MARK: Initializers
 
-    init(withModel model: CodeModel) {
+    required init(withModel model: CodeModel) {
         self.model = model
     }
 
     // MARK: Methods
 
     /// Begin naming process
+    // swiftlint:disable cyclomatic_complexity
     func process() throws {
-        // TODO: Fill in naming logic here
+        let schemas = model.schemas
```

Comment:
This is the bulk of the naming logic. Most of the information linked from `model.schemas` doesn't translate to customer-facing code. It simply will link internally to metadata.